File: SwordFingerOffer/python/jz_005.py
"""
 * @file     replaceSpace.py
 * @brief    https://leetcode-cn.com/problems/ti-huan-kong-ge-lcof/
 * @author   YongDu
 * @date     2021-08-18
"""


class Solution:
    def replaceSpace(self, s: str) -> str:
        return s.replace(' ', '%20')

File: SwordFingerOffer/cpp/jz_005.cpp
/**
 * @file     jz_005.cpp
 * @brief    https://leetcode-cn.com/problems/ti-huan-kong-ge-lcof/
 * @author   YongDu
 * @date     2021-09-09
 */

class Solution {
public:
  string replaceSpace(string s) {
    if (s.empty()) {
      return string();
    }
    std::string result;
    for (auto ch : s) {
      if (ch == ' ') {
        result += "%20";
      } else {
        result += ch;
      }
    }
    return result;
  }
};
